make/autoconf/flags-cflags.m4

```diff
@@ -566,6 +568,12 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_HELPER],
       TOOLCHAIN_CFLAGS_JDK="-pipe"
       TOOLCHAIN_CFLAGS_JDK_CONLY="-fno-strict-aliasing" # technically NOT for CXX
     fi
+
+    # Disable experimental isel due to a known issue in llvm-8, which generates wrong debug info.
+    # (see https://bugs.llvm.org/show_bug.cgi?id=40887)
```

Comment:
Does using the latest released LLVM 10.0.0 resolve this problem? If yes, we don't need to add this flag.

TSAN don't have a strong fixation to LLVM 8, and we'd recommend using the latest released LLVM. Version 8 just happened to be the latest released version when we wrote the wiki page. We are happy to update the Wiki page.

Reply:
I just tried llvm-9, the debug info is still wrong, makes debugging hard, even though the bug report said it has been fixed in llvm-9. after disable it by fno-experimental-isel, everything is fine.

Reply:
I will take a try to use clang-10.

Reply:
or I can remove the configuration, and use it only locally.

Reply:
```
Thread 3 "java" hit Breakpoint 2, 0x0000ffffb3953da4 in Arguments::set_heap_size () at /home/wave/workspace/jdk_tsan_workspace_aarch64/src/hotspot/share/oops/klass.hpp:297
297       ClassLoaderData* class_loader_data() const               { return _class_loader_data; }
(gdb) x/20i $pc
=> 0xffffb3953da4 <Arguments::set_heap_size()+12>:      mov     w0, #0x349                      // #841
   0xffffb3953da8 <Arguments::set_heap_size()+16>:      adrp    x8, 0xffffb49d7000
   0xffffb3953dac <Arguments::set_heap_size()+20>:      ldr     x8, [x8, #1592]
   0xffffb3953db0 <Arguments::set_heap_size()+24>:      adrp    x9, 0xffffb49d3000
   0xffffb3953db4 <Arguments::set_heap_size()+28>:      ldr     x9, [x9, #3264]
   0xffffb3953db8 <Arguments::set_heap_size()+32>:      adrp    x10, 0xffffb49d6000
   0xffffb3953dbc <Arguments::set_heap_size()+36>:      ldr     x10, [x10, #2112]
   0xffffb3953dc0 <Arguments::set_heap_size()+40>:      adrp    x11, 0xffffb49d6000
   0xffffb3953dc4 <Arguments::set_heap_size()+44>:      ldr     x11, [x11, #432]
   0xffffb3953dc8 <Arguments::set_heap_size()+48>:      adrp    x12, 0xffffb4a12000 <_ZL17special_jvm_flags+608>
   0xffffb3953dcc <Arguments::set_heap_size()+52>:      add     x12, x12, #0x28
   0xffffb3953dd0 <Arguments::set_heap_size()+56>:      adrp    x13, 0xffffb49d4000
   0xffffb3953dd4 <Arguments::set_heap_size()+60>:      ldr     x13, [x13, #1464]
   0xffffb3953dd8 <Arguments::set_heap_size()+64>:      adrp    x14, 0xffffb49d8000
   0xffffb3953ddc <Arguments::set_heap_size()+68>:      ldr     x14, [x14, #960]
   0xffffb3953de0 <Arguments::set_heap_size()+72>:      str     x8, [sp, #80]
   0xffffb3953de4 <Arguments::set_heap_size()+76>:      str     x9, [sp, #72]
   0xffffb3953de8 <Arguments::set_heap_size()+80>:      str     x10, [sp, #64]
   0xffffb3953dec <Arguments::set_heap_size()+84>:      str     x11, [sp, #56]
   0xffffb3953df0 <Arguments::set_heap_size()+88>:      str     x12, [sp, #48]
```
seems debug info by llvm-10 is still wrong, and assembly code is correct. I'm not sure if it's related to my env. but disable it by -fno-experimental-isel, everything is ok.

Reply:
Thanks for testing it out.